tests/pagination/tests.py

```diff
@@ -257,47 +257,51 @@ def setUp(self):
             a.save()
 
     def test_first_page(self):
+        p = None
         paginator = Paginator(Article.objects.all(), 5)
-        p = paginator.page(1)
-        self.assertEqual("<Page 1 of 2>", six.text_type(p))
-        self.assertQuerysetEqual(p.object_list, [
-            "<Article: Article 1>",
-            "<Article: Article 2>",
-            "<Article: Article 3>",
-            "<Article: Article 4>",
-            "<Article: Article 5>"
-        ],
-            ordered=False
-        )
-        self.assertTrue(p.has_next())
-        self.assertFalse(p.has_previous())
-        self.assertTrue(p.has_other_pages())
-        self.assertEqual(2, p.next_page_number())
-        with self.assertRaises(InvalidPage):
-            p.previous_page_number()
-        self.assertEqual(1, p.start_index())
-        self.assertEqual(5, p.end_index())
+        with self.assertRaises(RuntimeWarning):
```

Comment:
Yes, you are right.
The exception is raised when calling `paginator.page(1)` method, but I think it's much cleaner to run there 3 tests with ordered QuerySet.